scapy/arch/linux.py

```diff
@@ -146,7 +146,7 @@ def attach_filter(s, bpf_filter, iface):
     nb = int(lines[0])
     bpf = ""
     for l in lines[1:]:
-        bpf += struct.pack("HBBI",*map(long,l.split()))
+        bpf += struct.pack("HBBI", *[int(e) for e in l.split()])
```

Comment:
You don't need a list here: `*(int(e) for e in l.split())` should work just as well.